### ISSUE TITLE
xsl:on-completion Code Coverage Non-Conformance

### ISSUE THREAD

**Issue by birdya22** [xspec/xspec#1943]
### Current Non-Conformance
The `xslt-code-coverage-behavior-by-element.md` file says that `xsl:on-completion` elements should be marked based on their child elements.
The coverage-report.xsl in 3.0.3 marks them as 'missed' because there is no hit element in the XSpec trace file and there is no rule in coverage-report.xsl to process them.

### Changes in `coverage-report.xsl`
A rule needs to be added to `coverage-report.xsl` as follows for `xsl:on-completion`:
If node has no children, mark it as 'unknown'. 
If the trace data has a "hit" element for a child of this node, then mark this node as a 'hit'. 
Otherwise, mark it as 'missed'. 

### Changes in `test/end-to-end/cases-coverage/expected`
In the coverage report `xsl-iterate-01-coverage.html` 3 of the 4 `xsl:on-completion` elements will be marked as 'hit' rather than 'missed'. The one that will stay as 'missed' is the one with the  `<!-- Expected miss -->` comment on the right hand side.

**Comment by galtm** [comment]
@birdya22 : Looking into this issue as well as #1942, I wanted to run a couple of things by you to see if we have the same understanding.

1. The logic for "unknown" applies not only to the elements named in open GitHub issues that recommend the Use Child Data rule, but also the elements that already use the Use Child Data rule: `xsl:when`, `xsl:otherwise`, `xsl:matching-substring`, and `xsl:non-matching-substring`.

> If node has no children, mark it as 'unknown'.

2. In addition to the logic above: if element X has at least one child node, but all descendants are untraceable due to their markup alone (i.e., regardless of what the .xspec file covers), then we should mark X as 'unknown'. So, for example, `xsl:otherwise` would be marked 'unknown' in all of the following cases.
```
<xsl:otherwise/>
<xsl:otherwise><!-- untraceable child node --></xsl:otherwise>
<xsl:otherwise>
    <xsl:sequence select="'untraceable child node'"/>
</xsl:otherwise>
```

**Comment by birdya22** [comment]
Yes and Yes.

**Comment by galtm** [comment]
> > If node has no children, mark it as 'unknown'.
> 
> 2. In addition to the logic above: if element X has at least one child node, but all descendants are untraceable due to their markup alone (i.e., regardless of what the .xspec file covers), then we should mark X as 'unknown'. So, for example, `xsl:otherwise` would be marked 'unknown' in all of the following cases.
> 
> ```
> <xsl:otherwise/>
> <xsl:otherwise></xsl:otherwise>
> <xsl:otherwise>
>     <xsl:sequence select="'untraceable child node'"/>
> </xsl:otherwise>
> ```

I found a mistake in the "missed" logic for Use Child Data that I implemented in #1974. Suppose you have one descendant that's untraceable and one that's traceable but doesn't have a hit in the trace. The logic I implemented produces the status "missed". But if the code is
```
 <xsl:otherwise>
     <xsl:sequence select="error((),'untraceable child node')"/>
     <xsl:text>traceable but missed</xsl:text>
 </xsl:otherwise>
```
and `xsl:otherwise` is executed, "missed" is incorrect. I'll fix this and also recheck my thinking about the rules in #1978.
